### PR TITLE
[VL] Short decimal should memcpy int64 data to arrow int128

### PR DESCRIPTION
**Pull request by liujiayi771**
## What changes were proposed in this pull request?

https://github.com/oap-project/gluten/pull/1370 fixed the core dump error for decimal data. However, we found that the correctness of shuffle write is wrong. We cannot directly use SplitFixedType<uint64_t> method to handle short decimal. In velox short decimal is stored in int64, arrow always use int128 to store decimal, so we should copy int64 data to int128 dst.

## How was this patch tested?
Run 1T tpcds, and compare results.

### CODE REVIEW

**Comment by liujiayi771** [comment]
@FelixYBW From the perspective of the open source community, ARM adaptation is also a very important part of gluten. Is it possible to add a simple CI test to arm platform? Similar to the test of CentOS8. We can try to add an ARM CI test, use github resources or provide an arm machine to run the test.

File: cpp/velox/shuffle/VeloxShuffleWriter.cc [hunk 1]
```diff
@@ -502,7 +502,23 @@ arrow::Status VeloxShuffleWriter::SplitFixedWidthValueBuffer(const velox::RowVec
...
+            auto dst_pid_base = reinterpret_cast<uint32x4_t*>(partition_buffer_idx_offset_[pid]);
+            auto pos = partition_2_row_offset_[pid];
+            auto end = partition_2_row_offset_[pid + 1];
+            for (; pos < end; ++pos) {
+              auto row_id = row_offset_2_row_id_[pos];
+              const uint64_t value = reinterpret_cast<const uint64_t*>(src_addr)[row_id]; // copy
+              memcpy(dst_pid_base, &value, sizeof(uint64_t));
+              dst_pid_base += 1;
+            }
+          }
```
Comment (FelixYBW) [inline cpp/velox/shuffle/VeloxShuffleWriter.cc:521]: How velox handles the x86 vs. arm code? I'm thinking if it's a good way to add if defined in codes

**Comment by FelixYBW** [comment]
> @FelixYBW From the perspective of the open source community, ARM adaptation is also a very important part of gluten. Is it possible to add a simple CI test to ARM platform? Similar to the test of CentOS8. We can try to add an ARM CI test, use github resources or provide an ARM machine to run the test.

We used a powerful machine to do the CI tests. The github resources definitely aren't fast enough. The other big problem is most developers don't have a ARM development, once there is any bug we can hardly debug them, like this PR. What I'm suggested to AWS guys are that they run the CI tests from their side every night, once there is any issues, they should fix them on ARM by themselves. We are OK to merge the PRs

File: cpp/velox/shuffle/VeloxShuffleWriter.cc (hunk 1 above)
Comment (liujiayi771) [reply cpp/velox/shuffle/VeloxShuffleWriter.cc:521]: Currently, there is also if defined in the velox code to handle aarch64, but velox uses xsimd, which shields many differences between different platforms. In terms of int128, velox does not choose the vectorization type according to the platform. If shuffle write wants to use the vectorized type, it should be differentiated by platform, but maybe we can create a lower-level method to deal with the difference of int128.

**Review by FelixYBW — APPROVED** [review]

**Comment by FelixYBW** [comment]
@zzcclp Looks CK CI doesn't work again. 
Also the PR should skip the CK backend CI test. Looks our filter failed

**Comment by FelixYBW** [comment]
@liujiayi771 Ready to merge?

**Comment by liujiayi771** [comment]
> @liujiayi771 Ready to merge?

Yes, there are no other modifications, the test is correct.

**Pull request merged**